edx_sga/sga.py

```diff
@@ -39,7 +39,7 @@
 from xmodule.contentstore.content import StaticContent
 from xmodule.util.duedate import get_extended_due_date
 
-from edx_sga.constants import ITEM_TYPE
+from edx_sga.constants import ATTR_KEY_ANONYMOUS_USER_ID, ATTR_KEY_USER_IS_STAFF, ATTR_KEY_USER_ROLE, ITEM_TYPE
```

Comment:
@arslanashraf7, the goal is never to import from the `edx-platform` in external packages. We want the developers to be able to use the package and the test suite locally - i.e., independently from the devstack. This also complicates the CI, as the integration tests need something like [this](https://github.com/open-craft/section-to-course/blob/23b58ff/.github/workflows/integration-tests.yml) to run `edx-platform`. Therefore, introducing devstack as a dependency makes developing and testing much more complex.

We should treat these values as a stable API that will not be changed because other XBlocks (like [Drag and Drop v2](https://github.com/openedx/xblock-drag-and-drop-v2/blob/db6130a/drag_and_drop_v2/utils.py#L184)) depend on it, too. This has been in `edx-platform` for almost two years, so changing this approach without breaking the backward compatibility won't be easy. In the future, we could move these variables to a different repository (e.g., `XBlock` or `xblock-utils`) to import them into third-party packages. However, importing them would still be incompatible with previous releases, so we should consider if it's worth it first.

Reply:
I agree with this point of view and as you said they are least likely to be changes in the platform itself. That was the only hazard I had in mind for keeping the xBlock stable if the values change in edX and the xBlock works the same in that case. 

So we have the option of copying the fields with the assumption that these won't change in edX easily vs using them right from within the platform and keeping the coupling. Although we have been using other functionalities from edX within this and other xBlocks where we import certain things from edX.

In this case, I agree, If can keep these, Even if it breaks at some point we'll have to make a code change here as well.

@Agrendalath Is this ready for review again?

Reply:
Yes, Maybe we will remove that in a followup PR as well. The CI passes which should probably be enough for our bot to publish a new package.